### PR TITLE
Reference dataset catalogue page update

### PR DESCRIPTION
**Pull request by sekharpanja**
### Description of change
This PR:
    Updates reference dataset catalogue page to new design
    Adds columns and code snippets
    adds new overflow columns page
    adds new enhanced preview page
    allows data download as both csv and json

### Checklist

* [x] Have tests been added to cover any changes?

### CODE REVIEW

File: dataworkspace/dataworkspace/apps/datasets/views.py [hunk 1]
```diff
@@ -94,6 +94,7 @@
     find_dataset_or_visualisation_for_bookmark,
     get_code_snippets_for_table,
     get_code_snippets_for_query,
+    get_code_snippets_for_refereance_table,
```
Comment (markhigham) [inline dataworkspace/dataworkspace/apps/datasets/views.py:97]: `refereance`

File: dataworkspace/dataworkspace/static/data-grid.js [hunk 2]
```diff
@@ -279,6 +279,24 @@ function initDataGrid(columnConfig, dataEndpoint, records, exportFileName) {
     return;
   });
 
+  document.querySelector('#data-grid-json-download').addEventListener('click', function (e) {
+    let rowData = [];
```
Comment (markhigham) [inline dataworkspace/dataworkspace/static/data-grid.js:283]: `let` is only partially supported in IE11. This is probably OK but it may be worth testing

**Review by markhigham — CHANGES_REQUESTED** [review]
A couple of comments inline

File: dataworkspace/dataworkspace/apps/datasets/views.py [hunk 3]
```diff
@@ -1140,6 +1152,45 @@ def get(self, request, dataset_uuid, table_uuid):
...
+                "columns": columns,
+            },
+        )
+
+
+class ReferenceDatasetGridView(View):
+    def get(self, request, dataset_uuid):
+        try:
+            dataset = ReferenceDataset.objects.get(uuid=dataset_uuid)
+        except (DataSet.DoesNotExist, SourceTable.DoesNotExist):
```
Comment (niross) [inline dataworkspace/dataworkspace/apps/datasets/views.py:1182]: Should this be `ReferenceDataset.DoesNotExist`?

Comment (niross) [reply dataworkspace/dataworkspace/apps/datasets/views.py:1182]: Actually saying that could we just use `get_object_or_404`?

**Review by niross — APPROVED** [review]

**Review by michalc — APPROVED** [review]

**Pull request merged**